src/nix-collect-garbage/nix-collect-garbage.cc

```diff
@@ -43,11 +43,22 @@ void removeOldGenerations(std::string dir)
             }
             if (link.find("link") != std::string::npos) {
                 printInfo("removing old generations of profile %s", path);
-                if (deleteOlderThan != "") {
+                if (deleteOlderThan == "") {
+                    deleteOldGenerations(path, dryRun);
+                    continue;
+                }
+                if (deleteOlderThan.back() == 'd') {
```

Comment:
It relies implicitly on the format of `--delete-older-than` period and might break should that change. Might be a better option to try parsing time period first and fall back to number if failed, or use a separate param altogether.